src/olympia/addons/tests/test_views.py

```diff
@@ -3940,6 +3940,84 @@ class TestVersionViewSetUpdateJWTAuth(TestVersionViewSetUpdate):
     client_class = APITestClientJWT
 
 
+class TestVersionViewSetDelete(TestCase):
+    client_class = APITestClientSessionID
+
+    def setUp(self):
+        super().setUp()
+        self.user = user_factory(read_dev_agreement=self.days_ago(0))
+        self.addon = addon_factory(users=(self.user,))
+        self.version = self.addon.current_version
+        self.url = reverse_ns(
+            'addon-version-detail',
+            kwargs={'addon_pk': self.addon.slug, 'pk': self.version.id},
+            api_version='v5',
+        )
+
+    def test_delete(self):
+        assert not self.version.deleted
+
+        response = self.client.delete(self.url)
+        assert response.status_code == 401
+        assert not self.version.reload().deleted
+
+        self.client.login_api(self.user)
+        response = self.client.delete(self.url)
+        assert response.status_code == 204
+        assert self.version.reload().deleted
+        assert self.addon.reload().status == amo.STATUS_NULL
+
+    def test_not_author_cannot_delete(self):
+        another_user = user_factory(read_dev_agreement=self.days_ago(0))
+        self.client.login_api(another_user)
+        response = self.client.delete(self.url)
+        assert response.status_code == 403
+        assert not self.version.reload().deleted
+
+        # even if they are a reviewer
+        self.grant_permission(another_user, ':'.join(amo.permissions.ADDONS_EDIT))
```

Comment:
It's nice to test `ADDONS_EDIT` but we should also test with something that satisfies `acl.is_reviewer()`

Reply:
https://github.com/mozilla/addons-server/pull/20143/commits/9d099dd16899b0d939c7430db89e91095133fadc